### PR TITLE
ddl: fix panic tidb-server doesn't release table lock (#19586)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #19586 to release-4.0

---

### What problem does this PR solve?

Close https://github.com/pingcap/tidb/issues/19240

When `tidb-server` been `killed -9`(such as oom that killed by the system), the tidb-server won't release the held table locks, and this `dead` table lock will never be released unless user use `admin cleanup table lock` statement. 

### What is changed and how it works?

Add a background goroutine to periodic inspection the `dead` lock.

If the lock is held by the server that doesn't exist meta key (such as `DDLAllSchemaVersions` meta key) in ETCD, we think the server is dead, so we can clean the locks that hold by the dead tidb-server.

#### Why use `DDLAllSchemaVersions` meta key instead of `ServerInformationPath` meta key?

1. `Table lock` implementation with DDL, and `DDLAllSchemaVersions` meta key is a related DDL meta key.
2. `DDLAllSchemaVersions` TTL is 90s, but `ServerInformationPath` TTL is 10 min. The TTL`90s` is much better for usage.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM

### Release note 

- fix panic tidb-server doesn't release table lock

### CODE REVIEW

File: ddl/ddl.go [hunk 1]
```diff
@@ -258,14 +261,20 @@ func newDDL(ctx context.Context, options ...Option) *ddl {
...
 	if etcdCli := opt.EtcdCli; etcdCli == nil {
 		// The etcdCli is nil if the store is localstore which is only used for testing.
 		// So we use mockOwnerManager and MockSchemaSyncer.
 		manager = owner.NewMockManager(ctx, id)
 		syncer = NewMockSchemaSyncer()
 	} else {
 		manager = owner.NewOwnerManager(ctx, etcdCli, ddlPrompt, id, DDLOwnerKey)
+<<<<<<< HEAD
 		syncer = util.NewSchemaSyncer(etcdCli, id, manager)
+=======
```
Comment (zimulala) [inline ddl/ddl.go:274]: @crazycs520 
Please fix the conflict.

**Review by tangenta — COMMENTED** [review]
LGTM

**Review by AilinKid — COMMENTED** [review]
LGTM

**Review by lzmhhh123 — APPROVED** [review]

**Pull request merged**